File: backend/zvms/views/notices.py
from zvms.routelib import *
from zvms.res import AUTH
from zvms.views.structs import Notice, NoticeBody
import zvms.impls.notices

route(
    rule='/notices',
    method='GET',
    impl_func=zvms.impls.notices.search_notices
)

route(
    rule='/notices',
    method='POST',
    impl_func=zvms.impls.notices.send_notice,
    params=Notice,
    auth=AUTH.MANAGER | AUTH.TEACHER
)

route(
    rule='/notices/<int:id>',
    method='DELETE',
    impl_func=zvms.impls.notices.delete_notice,
    auth=AUTH.MANAGER | AUTH.TEACHER
)

route(
    rule='/notices/<int:id>',
    method='PUT',
    impl_func=zvms.impls.notices.update_notice,
    auth=AUTH.MANAGER | AUTH.TEACHER,
    params=NoticeBody
)

File: apis/paths/notices.ts
import { $ } from "zvms-apis-paths-gen";
import { authData as a } from "../users/index.js";
import { str ,structs} from "../types/index.js";

export default $(
    "/notices",
    [],
    {
        type: "GET",
        name: "search_notices"
    },
    {
        type: "POST",
        name: "send_notice",
        auths: [],
        req: structs.Notice
    },
    $(
        "/<int:id>",
        [],
        {
            type: "GET",
            name: "get_notice",
            desc: ""
        },
        {
            type: "DELETE",
            name: "delete_notice",
            auths: []
        },
        {
            type: "PUT",
            name: "update_notice",
            auths: []
        }
    )
);